pytorch_lightning/loggers/wandb.py

```diff
@@ -468,6 +498,45 @@ def after_save_checkpoint(self, checkpoint_callback: "ReferenceType[ModelCheckpo
         elif self._log_model is True:
             self._checkpoint_callback = checkpoint_callback
 
+    @staticmethod
+    @rank_zero_only
+    def download_artifact(
+        artifact: str,
+        save_dir: Optional[str] = None,
+        artifact_type: Optional[str] = None,
+        use_artifact: Optional[bool] = True,
+    ) -> str:
+        """Downloads an artifact from the wandb server.
+
+        Args:
+            artifact: The path of the artifact to download.
+            save_dir: The directory to save the artifact to.
+            artifact_type: The type of artifact to download.
+            use_artifact: Whether to add an edge between the artifact graph.
+
+        Returns:
+            The path to the downloaded artifact.
+        """
+        if wandb.run is not None and use_artifact:
+            artifact = wandb.run.use_artifact(artifact)
+        else:
+            api = wandb.Api()
+            artifact = api.artifact(artifact, type=artifact_type)
+
+        return artifact.download(root=save_dir)
+
+    def use_artifact(self, artifact: str, artifact_type: Optional[str] = None) -> "wandb.Artifact":
```

Comment:
Not sure if we should, but the arg name could be changed so that it matches that of wandb. It sounds a bit redundant to me as the method name is already `use_artifact`.
```suggestion
    def use_artifact(self, artifact: str, type: Optional[str] = None) -> "wandb.Artifact":
```

reference: https://cs.github.com/wandb/client?q=use_artifact

Reply:
The reason for using `artifact_type` was [this suggestion](https://github.com/PyTorchLightning/pytorch-lightning/pull/12764#discussion_r856346198)